ocaml/typing/typecore.ml

```diff
@@ -6626,11 +6649,12 @@ and type_expect_
             exp_attributes = sexp.pexp_attributes;
             exp_env = env }
   | Pexp_hole ->
-      begin match overwrite, Language_extension.is_enabled Overwriting with
-      | Overwriting(old_loc, typ, mode), true ->
+      begin match overwrite with
+      | Overwriting(old_loc, typ, fields_mode, _) ->
```

Comment:
I would expect needing to check the overwriting state here, to ensure that we're not looking for an allocation.